src/ripple/app/ledger/Ledger.cpp

```diff
@@ -914,7 +916,7 @@ Ledger::read (Keylet const& k) const
     if (! value)
         return nullptr;
     auto sle = std::make_shared<SLE>(
-        value->peekSerializer(), value->key());
+        SerialIter{value->data(), value->size()}, value->key());
```

Comment:
I added an overload to take the `SerialIter` by rvalue-reference.  The existing code ignored any bytes not consumed, so no change in behavior with respect to that.